drivers/sensor/bme680/bme680.c

```diff
@@ -0,0 +1,684 @@
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <kernel.h>
+#include <sensor.h>
+#include <init.h>
+#include <gpio.h>
+#include <misc/byteorder.h>
+#include <misc/__assert.h>
+
+#ifdef CONFIG_BME680_DEV_TYPE_I2C
+#include <i2c.h>
+#endif
+#include <logging/log.h>
+
+#include "bme680.h"
+
+#define LOG_LEVEL CONFIG_SENSOR_LOG_LEVEL
+LOG_MODULE_REGISTER(BME680);
+
+static int bme680_reg_read(struct BME680_data *data, u8_t start,
+			   u8_t *buf, int size)
+{
+#ifdef CONFIG_BME680_DEV_TYPE_I2C
+	return i2c_burst_read(data->i2c_master,
+			      data->i2c_slave_addr, start, buf, size);
+#endif
+	return 0;
+}
+
+static int bme680_reg_write(struct BME680_data *data, u8_t reg, u8_t val)
+{
+#ifdef CONFIG_BME680_DEV_TYPE_I2C
+	return i2c_reg_write_byte(data->i2c_master, data->i2c_slave_addr,
+				  reg, val);
+#endif
+	return 0;
+}
+
+/*
+ * Compensation code taken from Bosch BME680 driver
+ * https://github.com/BoschSensortec/BME680_driver/
+ *
+ * License agreement as follows:
+ *
+ * Copyright (C) 2017 - 2018 Bosch Sensortec GmbH
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * Redistributions of source code must retain the above copyright
+ * notice, this list of conditions and the following disclaimer.
+ *
+ * Redistributions in binary form must reproduce the above copyright
+ * notice, this list of conditions and the following disclaimer in the
+ * documentation and/or other materials provided with the distribution.
+ *
+ * Neither the name of the copyright holder nor the names of the
+ * contributors may be used to endorse or promote products derived from
+ * this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND
+ * CONTRIBUTORS "AS IS" AND ANY EXPRESS OR
+ * IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL COPYRIGHT HOLDER
+ * OR CONTRIBUTORS BE LIABLE FOR ANY
+ * DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY,
+ * OR CONSEQUENTIAL DAMAGES(INCLUDING, BUT NOT LIMITED TO,
+ * PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION)
+ * HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+ * WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ * ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE
+ *
+ * The information provided is believed to be accurate and reliable.
+ * The copyright holder assumes no responsibility
+ * for the consequences of use
+ * of such information nor for any infringement of patents or
+ * other rights of third parties which may result from its use.
+ * No license is granted by implication or otherwise under any patent or
+ * patent rights of the copyright holder.
```

Comment:
Non-Apache licensed code must go into the `ext/` folder

Reply:
This isn't at odds with the apache license, that is the final paragraph of the license which states use of the code doesn't grant the user any rights to use patents held by Bosch, it is a standard disclaimer.

Reply:
The project has a specific process for contributing non-apache licensed code. Please see https://docs.zephyrproject.org/latest/contribute/contribute_guidelines.html#contribute-non-apache for more details.

Adding a disclaimer to a standard license effectively makes a new license. If that new license is not OSI-approved, the board and TSC will most likely reject it.